app/code/Magento/Inventory/Model/ReservationBuilderInterface.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model;
+
+use Magento\InventoryApi\Api\Data\ReservationInterface;
+
+/**
+ * Used to instantiate ReservationInterface objects
+ *
+ * @see ReservationInterface
+ */
+interface ReservationBuilderInterface
+{
+    /**
+     * @param int|null $reservationId
+     * @return ReservationBuilder
+     */
+    public function setReservationId($reservationId): ReservationBuilder;
+
+    /**
+     * @param int $stockId
+     * @return ReservationBuilder
+     */
+    public function setStockId(int $stockId): ReservationBuilder;
+
+    /**
+     * @param string $sku
+     * @return ReservationBuilder
+     */
+    public function setSku(string $sku): ReservationBuilder;
+
+    /**
+     * @param float $quantity
+     * @return ReservationBuilder
+     */
+    public function setQuantity(float $quantity): ReservationBuilder;
+
+    /**
+     * @param string|null $metadata
```

Comment:
don't think that we need to call `setMetadata` passing null `value`

thus, the only possible input param would be `string`

Reply:
>don't think that we need to call setMetadata passing null value

should this also be applied to `setReservationId()`?

Reply:
@aleron75 good catch! you are right